r/tests/testthat/test-dplyr-string-functions.R

```diff
@@ -467,6 +467,25 @@ test_that("strsplit and str_split", {
   )
 })
 
+test_that("str_to_lower, str_to_upper, and str_to_title", {
+  df <- tibble(x = c("1Foo1", " \tB a R\n", "!apACHe aRroW!"))
+  funcs <- c(str_to_lower, str_to_upper, str_to_title)
+  for (func in funcs) {
+    expect_dplyr_equal(
+      input %>%
+        transmute(x = func(x)) %>%
```

Comment:
This should work, but because of how these tests are evaluated, I think something is getting lost leading to these errors: https://github.com/apache/arrow/pull/11232/checks?check_run_id=3748190500#step:8:17492

Reply:
In other places, we have tested lists of functions like these by transmuting/mutating multiple columns https://github.com/apache/arrow/blob/master/r/tests/testthat/test-dplyr.R#L996-L1009

something like:

```
expect_dplyr_equal(
       input %>%
         transmute(
           x_lower = str_to_lower(x),
           x_upper = str_to_upper(x),
           x_title = str_to_title(x)
         ) %>%
         collect()
```